src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterApplicationBuilderExtensions.cs

```diff
@@ -21,38 +21,133 @@
+        {
+            Guard.ThrowIfNull(predicate);
+            return UseOpenTelemetryPrometheusScrapingEndpoint(app, meterProvider: null, predicate: predicate, path: null, configureBranchedPipeline: null);
+        }
+
+        /// <summary>
+        /// Adds OpenTelemetry Prometheus scraping endpoint middleware to an
+        /// <see cref="IApplicationBuilder"/> instance.
+        /// </summary>
+        /// <remarks>Note: A branched pipeline is created based on the <paramref
+        /// name="predicate"/> or <paramref name="path"/>. If neither <paramref
+        /// name="predicate"/> nor <paramref name="path"/> are provided then
+        /// <see cref="PrometheusExporterOptions.ScrapeEndpointPath"/> is
+        /// used.</remarks>
         /// <param name="app">The <see cref="IApplicationBuilder"/> to add
         /// middleware to.</param>
         /// <param name="meterProvider">Optional <see cref="MeterProvider"/>
         /// containing a <see cref="PrometheusExporter"/> otherwise the primary
         /// SDK provider will be resolved using application services.</param>
-        /// <returns>A reference to the <see cref="IApplicationBuilder"/> instance after the operation has completed.</returns>
-        public static IApplicationBuilder UseOpenTelemetryPrometheusScrapingEndpoint(this IApplicationBuilder app, MeterProvider meterProvider = null)
+        /// <param name="predicate">Optional predicate for deciding if a given
+        /// <see cref="HttpContext"/> should be branched. If supplied <paramref
+        /// name="path"/> is ignored.</param>
+        /// <param name="path">Optional path to use for the branched pipeline.
+        /// Ignored if <paramref name="predicate"/> is supplied.</param>
+        /// <param name="configureBranchedPipeline">Optional callback to
+        /// configure the branched pipeline. Called before registration of the
+        /// Prometheus middleware.</param>
+        /// <returns>A reference to the original <see
+        /// cref="IApplicationBuilder"/> for chaining calls.</returns>
+        public static IApplicationBuilder UseOpenTelemetryPrometheusScrapingEndpoint(
+            this IApplicationBuilder app,
+            MeterProvider meterProvider,
+            Func<HttpContext, bool> predicate,
+            string path,
+            Action<IApplicationBuilder> configureBranchedPipeline)
         {
-            var options = app.ApplicationServices.GetOptions<PrometheusExporterOptions>();
+            // Note: Order is important here. MeterProvider is accessed before
+            // GetOptions<PrometheusExporterOptions> so that any changes made to
+            // PrometheusExporterOptions in deferred AddPrometheusExporter
+            // configure actions are reflected.
+            meterProvider ??= app.ApplicationServices.GetRequiredService<MeterProvider>();
```

Comment:
Is it possible for meterProvider to be null after this line - i.e., no meter provider explicitly passed and no meter provider registered? If so, what happens? Do we fail initialization?

Reply:
`GetRequiredService<T>` throws if no service can be resolved. It won't ever be null.